hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDFSPermission.java

```diff
@@ -260,6 +261,33 @@ private void createAndCheckPermission(OpType op, Path name, short umask,
     checkPermission(name, expectedPermission, delete);
   }
 
+  @Test
+  public void testFSNamesystemCheckAccess() throws Exception {
+    Path testValidDir = new Path("/test1");
+    Path testValidFile = new Path("/test1/file1");
+    Path testInvalidPath = new Path("/test2");
+    fs = FileSystem.get(conf);
+
+    fs.mkdirs(testValidDir);
+    fs.create(testValidFile);
+
+    fs.access(testValidDir, FsAction.READ);
+    fs.access(testValidFile, FsAction.READ);
+
+    assertTrue(fs.exists(testValidDir));
+    assertTrue(fs.exists(testValidFile));
+
+    try {
+      fs.access(testInvalidPath, FsAction.READ);
+      fail("Failed to get expected FileNotFoundException");
+    } catch (FileNotFoundException e) {
+      GenericTestUtils.assertExceptionContains(
+              "Path not found: " + testInvalidPath, e);
+    } finally {
+      fs.delete(testValidDir, true);
+    }
+  }
+
```

Comment:
This is like testing normal fs.access also which isn't required, we just changed the exception, we can test that only.
Can use LambdaTestUtils for that rather than the present try-catch. Something like this should do:
```
  @Test
  public void testFSNamesystemCheckAccess() throws Exception {
    Path testInvalidPath = new Path("/test2");
    fs = FileSystem.get(conf);

    LambdaTestUtils.intercept(FileNotFoundException.class,
        "Path not found: " + testInvalidPath,
        () -> fs.access(testInvalidPath, FsAction.READ));
  }
```

Reply:
clearer than before, thanks @ayushtkn